src/webgpu/api/operation/command_buffer/image_copy.spec.ts

```diff
@@ -627,6 +629,250 @@ class ImageCopyTest extends GPUTest {
+    const uniformBufferSize = kMinDynamicBufferOffsetAlignment * (stencilBitCount - 1) + 4;
+    const uniformBuffer = this.device.createBuffer({
+      size: uniformBufferSize,
+      usage: GPUBufferUsage.COPY_DST | GPUBufferUsage.UNIFORM,
+    });
+    const uniformBufferData = new Uint32Array(uniformBufferSize / 4);
+    for (let i = 1; i < stencilBitCount; ++i) {
+      uniformBufferData[(kMinDynamicBufferOffsetAlignment / 4) * i] = i;
+    }
+    this.queue.writeBuffer(uniformBuffer, 0, uniformBufferData);
+
+    // Prepare the base render pipeline descriptor (all the settings expect stencilReadMask).
+    const bindGroupLayout = this.device.createBindGroupLayout({
+      entries: [
+        {
+          binding: 0,
+          visibility: GPUShaderStage.FRAGMENT,
+          buffer: {
+            type: 'uniform',
+            minBindingSize: 4,
+            hasDynamicOffset: true,
+          },
+        },
+        {
+          binding: 1,
+          visibility: GPUShaderStage.FRAGMENT,
+          storageTexture: {
+            access: 'read-only',
+            format: referenceTextureFormat,
+          },
+        },
+      ],
+    });
+    const renderPipelineDescriptorBase: GPURenderPipelineDescriptor = {
+      layout: this.device.createPipelineLayout({ bindGroupLayouts: [bindGroupLayout] }),
+      vertex: {
+        module: this.device.createShaderModule({
+          code: `
+            [[stage(vertex)]]
+            fn main([[builtin(vertex_index)]] VertexIndex : u32)-> [[builtin(position)]] vec4<f32> {
+              var pos : array<vec2<f32>, 6> = array<vec2<f32>, 6>(
+                  vec2<f32>(-1.0,  1.0),
+                  vec2<f32>(-1.0, -1.0),
+                  vec2<f32>( 1.0,  1.0),
+                  vec2<f32>(-1.0, -1.0),
+                  vec2<f32>( 1.0,  1.0),
+                  vec2<f32>( 1.0, -1.0));
+              return vec4<f32>(pos[VertexIndex], 0.0, 1.0);
+            }`,
+        }),
+        entryPoint: 'main',
+      },
+
+      fragment: {
+        module: this.device.createShaderModule({
+          code: `
+            [[block]] struct Params {
+              stencilBitIndex: i32;
+            };
+            [[group(0), binding(0)]] var<uniform> param: Params;
+            [[group(0), binding(1)]] var storageImage: texture_storage_2d<rgba8unorm, read>;
+            [[stage(fragment)]]
+            fn main() -> [[location(0)]] vec4<f32> {
+              return textureLoad(storageImage, vec2<i32>(param.stencilBitIndex, 0));
+            }`,
+        }),
+        entryPoint: 'main',
+        targets: [
+          {
+            // As we implement "rendering one bit in each draw() call" with blending operation
+            // 'add', the format of outputTexture must support blending.
+            format: 'r8unorm',
+            blend: {
+              color: { srcFactor: 'one', dstFactor: 'one', operation: 'add' },
+              alpha: { srcFactor: 'one', dstFactor: 'one', operation: 'add' },
```

Comment:
nit: I think we might be able to skip alpha since we're r8unorm that has no alpha channel.

Reply:
Oh we cannot skip setting alpha as GPUBlendState.alpha is not optional.

Reply:
That's right. You can set it to `{}` though (if you want).